Fix autoawq gemma convert

dengzheng-cloud opened the pull request:
when using autoawq to quantize gemma-2b-it model, output model-00001-of-00002.safetensors will include lm_head.weight, this pr could prevent convert error happening in

`python convert-hf-to-gguf.py ${quantized_path} --outfile model.gguf`

fix #6633

dengzheng-cloud committed:
fix autoawq quantized gemma model convert error

using autoawq to quantize gemma model will include a lm_head.weight tensor in model-00001-of-00002.safetensors. it result in this situation that convert-hf-to-gguf.py can't map lm_head.weight. skip loading this tensor could prevent this error.

dengzheng-cloud committed:
Merge branch 'ggerganov:master' into fix-awq-gemma-convert

ggerganov commented on convert-hf-to-gguf.py:
```diff
@@ -2277,6 +2277,11 @@ def write_tensors(self):
         for name, data_torch in self.get_tensors():
+            # lm_head is not used in llama.cpp, while autoawq will include this tensor in model
+            # To prevent errors, skip loading lm_head.weight.
+            if "lm_head.weight" in name:
+                    continue
```
Let's change this to full string match and also print a short message that the tensor is being skipped

dengzheng-cloud replied:
already done in a new commit

dengzheng-cloud committed:
change code to full string match and print necessary message

change code to full string match and print a short message to inform users that lm_head.weight has been skipped.
